ui/calendar/date.js

```diff
@@ -17,40 +36,47 @@ var weekdays = [ "sun", "mon", "tue", "wed", "thu", "fri", "sat" ],
 		"sat": 6
 	};
 
-Globalize.locale( "en" );
-
-$.date = function( date, globalFormat ) {
-	if ( !( this instanceof $.date ) ) {
-		return new $.date( date, globalFormat );
+_Date = function( date, attributes ) {
+	if ( !( this instanceof _Date ) ) {
+		return new _Date( date, options );
 	}
+
+	this.setAttributes( attributes );
+
 	if ( typeof date === "string" && date.length ) {
-		this.dateObject = Globalize.parseDate( date, globalFormat );
+		this.dateObject = attributes.parse( date );
+	} else if ( $.type( date ) === "date" ) {
+		this.dateObject = date;
 	}
 
 	this.dateObject = this.dateObject || new Date();
-	this.globalFormat = globalFormat;
+	this.selected = null;
 };
 
-$.date.prototype = {
-	setFormat: function( format ) {
-		if ( format ) {
-			this.globalFormat = format;
-		}
-		return this;
+$.extend( _Date.prototype, {
+
+	setAttributes: function( attributes ) {
+		this.attributes = attributes;
+		this.firstDay = weekdaysRev[ this.attributes.firstDay ];
 	},
+
 	//TODO: same as the underlying Date object's terminology, but still misleading.
 	//TODO: We can use .setTime() instead of new Date and rename to setTimestamp.
 	setTime: function( time ) {
 		this.dateObject = new Date( time );
 		return this;
 	},
+
 	setDay: function( day ) {
 		var date = this.dateObject;
+		// FIXME: Why not to use .setDate?
 		this.dateObject = new Date( date.getFullYear(), date.getMonth(), day, date.getHours(),
 			date.getMinutes(), date.getSeconds() );
 		return this;
```

Comment:
Not sure if this was needed on order to prevent referencing of the new variable.